src/test/java/com/mavenit/argos/pages/ResultsPage.java

```diff
@@ -0,0 +1,86 @@
+package com.mavenit.argos.pages;
+
+import com.mavenit.argos.driver.DriverFactory;
+import com.mavenit.argos.utils.Helpers;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ResultsPage extends DriverFactory {
+
+    public String getSearchTitle() {
+        return driver.findElement(By.className("search-title__term")).getText();
+    }
+
+    public List<String> getAllProductNames() {
+        List<String> productNamesList = new ArrayList<>();
+        List<WebElement> productWebElements = isProductsAvailable();
+        for (WebElement indProduct : productWebElements) {
+            String actual = indProduct.getText();
+            productNamesList.add(actual);
+        }
+        return productNamesList;
+    }
+
+    public List<String> getAllProductRatings() {
+        List<String> productRatingList = new ArrayList<>();
+        List<WebElement> productWebElements = isProductRatingAvailable();
+        for (WebElement indProduct : productWebElements) {
```

Comment:
Is this method will wait for product to refresh after you apply rating filter ?
If yes good 
If no then you will get previous results before rating filter applied

Reply:
yes there is a wait for 4 to 5 seconds for the filters to apply, so that webdriver can read filtered results.